src/main/java/cube/model/sale/Sale.java

```diff
@@ -6,45 +6,54 @@
 package cube.model.sale;
 
 import java.util.Date;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 
-public class Sale {
-
+public class Sale implements Comparable<Sale>{
+	@JsonProperty
 	protected int quantitySold;
+	@JsonProperty
 	protected double profit;
+	@JsonProperty
 	protected double revenue;
+	@JsonProperty
 	protected Date soldDate;
 
-	public Sale() {
+	public Sale(){
 		this(0, 0, 0, null);
 	}
 
 	/**
 	 * Constructor with four arguments.
 	 *
-	 * @param quantity The quantity sold in this sale.
+	 * @param quantitySold The quantitySold sold in this sale.
 	 * @param profit The profit earned in this sale.
 	 * @param revenue The revenue earned in this sale.
-	 * @param date The date of sale.
+	 * @param soldDate The date of sale.
 	 */
-	public Sale(int quantity, double revenue, double profit, Date date) {
-		this.quantitySold = quantity;
+	public Sale(int quantitySold, double revenue, double profit, Date soldDate) {
+		this.quantitySold = quantitySold;
 		this.revenue = revenue;
 		this.profit = profit;
-		this.soldDate = date;
+		this.soldDate = soldDate;
 	}
 
+	@JsonIgnore
 	public int getQuantity() {
 		return quantitySold;
 	}
 
+	@JsonIgnore
 	public double getRevenue() {
 		return revenue;
 	}
 
+	@JsonIgnore
 	public double getProfit() {
 		return profit;
 	}
 
+	@JsonIgnore
```

Comment:
Json library limitations can't retrieve variables properly if getters are of different names causing the bug.